src/Proto.Actor/IMailbox.cs

```diff
@@ -77,27 +78,31 @@ public class DefaultMailbox : IMailbox
     {
         private readonly IMailboxQueue _systemMessages;
         private readonly IMailboxQueue _userMailbox;
+        private readonly IMailboxStatistics[] _stats;
         private IDispatcher _dispatcher;
         private IMessageInvoker _invoker;
 
         private int _status = MailboxStatus.Idle;
         private bool _suspended;
 
-        public DefaultMailbox(IMailboxQueue systemMessages, IMailboxQueue userMailbox)
+        public DefaultMailbox(IMailboxQueue systemMessages, IMailboxQueue userMailbox, params IMailboxStatistics[] stats)
         {
             _systemMessages = systemMessages;
             _userMailbox = userMailbox;
+            _stats = stats;
```

Comment:
foreach doesn't get much magic compiler love AFAIK. I think ensuring it is always non-null and readonly might be good - the JITter should be able to get rid of the internal null dereference check, right? And switching to a for-loop instead of a foreach-loop is always faster. Will update.